cfu-service/src/splitter.rs

```diff
@@ -0,0 +1,205 @@
+pub trait Customization {
+    /// Decides which firmware version to use based on the provided versions from all devices.
+    fn resolve_fw_versions(&self, versions: &[GetFwVersionResponse]) -> GetFwVersionResponse;
+
+    /// Decides which offer response to send based on the provided responses from all devices.
+    fn resolve_offer_response(&self, offer_responses: &[FwUpdateOfferResponse]) -> FwUpdateOfferResponse;
+
+    /// Decides which content response to send based on the provided responses from all devices.
+    fn resolve_content_response(&self, content_responses: &[FwUpdateContentResponse]) -> FwUpdateContentResponse;
+}
+
+/// Splitter struct
+pub struct Splitter<'a, C: Customization> {
+    /// CFU device
+    cfu_device: CfuDevice,
+    /// Component ID for each individual device
+    devices: &'a [ComponentId],
+    /// Customization for the Splitter
+    customization: C,
+}
+
+/// Maximum number of devices supported
+pub const MAX_SUPPORTED_DEVICES: usize = 8;
+
+impl<'a, C: Customization> Splitter<'a, C> {
+    /// Create a new Splitter
+    pub fn new(component_id: ComponentId, devices: &'a [ComponentId], customization: C) -> Self {
+        Self {
+            cfu_device: CfuDevice::new(component_id),
+            devices,
+            customization,
+        }
+    }
+
+    /// Create a new invalid FW version response
+    fn create_invalid_fw_version_response(&self) -> InternalResponseData {
+        let dev_inf = FwVerComponentInfo::new(FwVersion::new(0xffffffff), self.cfu_device.component_id());
+        let comp_info: [FwVerComponentInfo; MAX_CMPT_COUNT] = [dev_inf; MAX_CMPT_COUNT];
+        InternalResponseData::FwVersionResponse(GetFwVersionResponse {
+            header: GetFwVersionResponseHeader::new(1, GetFwVerRespHeaderByte3::NoSpecialFlags),
+            component_info: comp_info,
+        })
+    }
+
+    /// Create an offer rejection response
+    fn create_offer_rejection() -> InternalResponseData {
+        InternalResponseData::OfferResponse(FwUpdateOfferResponse::new_with_failure(
+            HostToken::Driver,
+            OfferRejectReason::InvalidComponent,
+            OfferStatus::Reject,
+        ))
+    }
+
+    /// Create a content rejection response
+    fn create_content_rejection(sequence: u16) -> InternalResponseData {
+        InternalResponseData::ContentResponse(FwUpdateContentResponse::new(
+            sequence,
+            CfuUpdateContentResponseStatus::ErrorInvalid,
+        ))
+    }
+
+    /// Process a fw version request
+    async fn process_get_fw_version(&self) -> InternalResponseData {
+        let mut versions = [GetFwVersionResponse {
+            header: Default::default(),
+            component_info: Default::default(),
+        }; MAX_SUPPORTED_DEVICES];
+
+        if self.devices.len() > MAX_SUPPORTED_DEVICES {
+            error!("More devices than supported");
+            return self.create_invalid_fw_version_response();
+        }
```

Comment:
How about checking for max supported devices before allocating stack space for all devices?